hera_sim/simulate.py

```diff
@@ -434,4 +454,93 @@ def add_xtalk(self, model='gen_whitenoise_xtalk', bls=None, **kwargs):
                 vis=self.data.data_array[blt_ind, 0, :, pol_ind],
                 xtalk=xtalk
             )
+    
+    
+    def run_sim(self, sim_file=None, **sim_params):
+        """
+        Accept a dictionary or YAML file of simulation parameters and add in
+        all of the desired simulation components to the Simulator object.
+
+        Args:
+            sim_file (str, optional): string providing a hook to a YAML file
+            
+            sim_params (dict, optional): dictionary of simulation parameters.
+                it must take the form {model:params}, where model is
+                a string specifying the simulation component to be added
+                and params is a dictionary providing all the keyword
+                arguments and their desired values.
+
+            One (and *only* one) of the above arguments must be provided. If 
+            *both* sim_file and sim_params are provided, then this function
+            will raise an AssertionError.
+        """
+
+        # keep track of which components don't use models
+        uses_no_model = []
+        for key, val in self.SIMULATION_COMPONENTS.items():
+            func = getattr(self, val)
+            # raise a NotImplementedError if using Py2
+            if sys.version_info.major < 3 or \
+               sys.version_info.major > 3 and sys.version_info.minor < 4:
+                raise VersionError("Please use a version of Python >= 3.4.")
```

Comment:
Do you think it's better to drop Py2 support entirely, or just manually define the entries in `uses_no_model` in the case that someone is running a version < 3.4?

Reply:
Nope, just drop it entirely for this function, I reckon.